### PR TITLE
[PIN-2282] Removed institution: retrieving institution name from tenant

### PR DESCRIPTION
**Pull request by bloodsky**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/purposeprocess/api/converters/partymanagement/OrganizationConverter.scala [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+import it.pagopa.interop.tenantmanagement.client.model.Tenant
 import it.pagopa.interop.purposeprocess.model.Organization
 
 import java.util.UUID
 
 object OrganizationConverter {
-  def dependencyToApi(id: UUID, organization: DependencyInstitution): Organization =
-    Organization(id = id, name = organization.description)
+  def dependencyToApi(id: UUID, tenant: Tenant): Organization =
+    Organization(id = id, name = tenant.externalId.origin)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/purposeprocess/api/converters/partymanagement/OrganizationConverter.scala]:
```suggestion
    Organization(id = id, name = tenant.name)
```

File: src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeVersionActivation.scala [hunk 2]
```diff
@@ -162,10 +162,8 @@ final case class PurposeVersionActivation(
   }
 
   def getTenantDescription(tenantId: UUID)(implicit contexts: Seq[(String, String)]): Future[String] = for {
-    tenant      <- tenantManagementService.getTenant(tenantId)
-    selfcareId  <- tenant.selfcareId.toFuture(MissingSelfcareId)
-    institution <- partyManagementService.getInstitutionById(selfcareId)
-  } yield institution.description
+    tenant <- tenantManagementService.getTenant(tenantId)
+  } yield tenant.name
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeVersionActivation.scala:166]: `tenantManagementService.getTenant(tenantId).map(_.name)`

I would also rename the function to `getTenantName` for clarity

File: src/test/scala/it/pagopa/interop/purposeprocess/SpecData.scala [hunk 3]
```diff
@@ -43,28 +43,16 @@ object SpecData {
...
-    taxCode = "taxCode",
-    attributes = Seq.empty,
-    zipCode = "00000",
-    institutionType = "PUBLIC"
-  )
-
   val tenant: Tenant = Tenant(
     UUID.randomUUID(),
-    selfcareId = institution.id.toString.some,
+    selfcareId = UUID.randomUUID.toString().some,
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposeprocess/SpecData.scala]:
```suggestion
    selfcareId = UUID.randomUUID.toString.some,
```

File: src/test/scala/it/pagopa/interop/purposeprocess/SpecHelper.scala [hunk 4]
```diff
@@ -331,8 +325,8 @@ trait SpecHelper extends SprayJsonSupport with DefaultJsonProtocol with MockFact
     val actualEService = eService.getOrElse(SpecData.eService).copy(descriptors = Seq(descriptor))
     mockAgreementsRetrieve(purpose.eserviceId, purpose.consumerId, Seq(agreement))
     mockEServiceRetrieve(purpose.eserviceId, actualEService)
-    mockOrganizationRetrieve(actualEService.producerId)
-    mockOrganizationRetrieve(purpose.consumerId)
+    mockOrganizationRetrieve(agreement.producerId)
+    mockOrganizationRetrieve(agreement.consumerId)
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposeprocess/SpecHelper.scala:329]: is there any reason for this change?

**Review by galales — CHANGES_REQUESTED** [review]

File: src/test/scala/it/pagopa/interop/purposeprocess/SpecHelper.scala (hunk 4 above)
Comment (bloodsky) [reply src/test/scala/it/pagopa/interop/purposeprocess/SpecHelper.scala:329]: I answer directly here, the different changes being connected. I originally made your first suggested change but the tests didn't pass. While trying to figure out the behavior I forgot about this change. The only way to pass producerId and consumerId correctly was to modify the mock. With your second suggested change now it's all resolved, thank! 😄

File: src/test/scala/it/pagopa/interop/purposeprocess/SpecData.scala (hunk 3 above)
Comment (bloodsky) [reply src/test/scala/it/pagopa/interop/purposeprocess/SpecData.scala]: This was the exact problem with my tests. UUID was invalid and in my mind that was related to the way I made changes inside the codebase (when I was passing tenant.name inside Organization). Thanks again!

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Review by TonioGela — APPROVED** [review]

**Pull request merged**